realm-elasticsearch/lib/realm/elasticsearch/repository.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'yaml'
+require 'active_support/core_ext/string'
+
+module Realm
+  module Elasticsearch
+    class Repository
+      def self.repo_name(value = :not_provided)
+        @repo_name = value.to_sym unless value == :not_provided
+        @repo_name = name.demodulize.underscore unless defined?(@repo_name)
+        @repo_name
+      end
+
+      def self.index_name(value = :not_provided)
+        @index_name = value.to_sym unless value == :not_provided
+        @index_name = repo_name.pluralize unless defined?(@index_name)
+        @index_name
+      end
+
+      def initialize(client)
+        @client = client
+      end
+
+      def find(id:)
+        single(client.get(index: index_name, id: id))
+      rescue ::Elasticsearch::Transport::Transport::Errors::NotFound
+        nil
+      end
+
+      def all
+        multiple(raw_search(query: { match_all: {} }))
+      end
+
+      def create(id: nil, **attrs)
+        client.index(index: index_name, type: '_doc', id: id, body: attrs, refresh: refresh?)
+      rescue ::Elasticsearch::Transport::Transport::Errors::Conflict
+        raise Realm::Persistence::Conflict
+      end
+
+      def update(id:, **attrs)
+        raw_update(id, doc: attrs)
+      end
+
+      def upsert(id:, **attrs)
+        raw_update(id, doc: attrs, doc_as_upsert: true)
+      end
+
+      def delete(id:)
+        client.delete(index: index_name, type: '_doc', id: id, refresh: refresh?)
+      rescue ::Elasticsearch::Transport::Transport::Errors::NotFound
+        # ignore
```

Comment:
Anti-pattern when rescuing without any result. This is dangerous because you can hide any errors.

Reply:
Well, I can log it somewhere as warning but it's really something we expect in event driven system with at least one delivery - you can have some message picked up multiple times occasionally.

Reply:
On the other hand I kind of follow the SQL logic. There you also don't throw error when you attempt to delete not existing records.

Reply:
@davidsevcik In Ar when you are trying to do some staff without `!` it will return the result true/false without the exception. Here you will get `nil` which means nothing. In my opinion(or not even mine) we should return something different than `nil`(maybe false??). We use the old rubocop, but in the new versions setting empty rescue block is the validation of rules and hiding the exceptions. 
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/SuppressedException

Reply:
OK, good point. I will return false. Thanks.